pkg/pillar/types/errortime.go

```diff
@@ -18,6 +18,25 @@ type ErrorAndTime struct {
 	ErrorTime time.Time
 }
 
+// SetOrAppendError - Add or Append an error to ErrorAndTime
+func (etPtr *ErrorAndTime) SetOrAppendError(et ErrorAndTime) {
+	if etPtr.HasError() {
+		// Error Alread Set. Append errors
+		// This is a simplistic way to carry all the errors. If we see this
+		// happening more and more, maintain an array of ErrorAndTime
+		if et.HasError() {
+			// Error Already set. Append errors
+			etPtr.Error = etPtr.Error + " \n " + et.Error
+			// Retain the timestamp of first error
+		}
+		// Else - new et doesn't have an error. etPtr already has an error.
+		//  Ignore et
```

Comment:
If a previous test set an error, then 5 minutes later or more the test succeeds, don't we hit this "else" and fail to clear the error?

Have you tested that case?

SetOrUpdateFromMap() calls this function indirectly hence the updating seems to always append error hence they are never cleared.

Reply:
As discussed on Slack:

func (portConfig *DevicePortConfig) UpdatePortStatusFromIntfStatusMap(
	intfErrMap IntfStatusMap) {
	for indx := range portConfig.Ports {
		portPtr := &portConfig.Ports[indx]
		et, ok := intfErrMap.StatusMap[portPtr.IfName]
		if ok {
			portPtr.ErrorAndTime = et
		}
		// Else - Port not tested. Retain the ErrorAndTime on the port.
	}
}
This is the one which actually Updates the Error in DPCfor each port
2:58
So - VerifyDeviceNetworkStatus --> This always returns a fresh set of errors - Irrespective of what happened before.
2:58
So.. Errors get appened only “With-in” a testing cycle.
2:59
And then we take the errors - and just store these for each port - aka.. we overwrite the existing errors with these new ones..
3:00
As the comment indicates - If the IntfStatusMap doesn’t not have an interface entry - it retains the existing error / state (edited) 
3:02
I tested this with making eth0 a management port ( Eth3 is the properly connected one )  and eth3 app-direct
I verified the device fellback to previous config - and eth0 has the error..
waited for 10 mins to make sure error is not cleared..
The corrected the config back - verified error goes away